cmd/lakectl/cmd/diff.go

```diff
@@ -109,13 +110,16 @@ func printDiffBranch(ctx context.Context, client api.ClientWithResponsesInterfac
 
 func printDiffRefs(ctx context.Context, client api.ClientWithResponsesInterface, left, right *uri.URI, twoDot bool) {
 	diffs := make(chan api.Diff, maxDiffPageSize)
-	err := diff.StreamRepositoryDiffs(ctx, client, left, right, "", diffs, twoDot)
-	if err != nil {
-		DieErr(err)
-	}
+	var wg errgroup.Group
+	wg.Go(func() error {
+		return diff.StreamRepositoryDiffs(ctx, client, left, right, "", diffs, twoDot)
+	})
```

Comment:
If `diff.StreamRepositoryDiffs` will create/close the channel, process the diff using goroutine inside the func - the use of this call will not require all the callers to `go` each time we use it.

Reply:
Deferring this comment to another issue, as we needed to also be able to handle errors that resulted from this function call.